roxie/ccd/ccdquery.cpp

```diff
@@ -1121,7 +1121,7 @@ class CQueryFactory : public CInterface, implements IQueryFactory, implements IR
                                     if (indexName)
                                     {
                                         bool isOpt = pretendAllOpt || node.getPropBool("att[@name='_isIndexOpt']/@value");
-                                        const IResolvedFile *indexFile = package.lookupFileName(indexName, isOpt, true, true, wu);
+                                        const IResolvedFile *indexFile = package.lookupFileName(indexName, isOpt, true, true, wu, !isDynamic);
```

Comment:
We are inside a if (!dynamic) - I think the code would be clearer if you passed true here